### ISSUE TITLE
Warnings in cuda builds: Kokkos_View.hpp(1750): warning: calling a __host__ function from a __host__ __device__ function is not allowed

### ISSUE THREAD

**Issue by ldh4** [kokkos/kokkos#5587]
#5491 caused
```
warning: calling a __host__ function from a __host__ __device__ function is not allowed
```

_Originally posted by @dalg24 in https://github.com/kokkos/kokkos/pull/5491#discussion_r1001113082_

**Comment by dalg24** [comment]
The issue is also present in the release candidate 3.7.01

**Comment by ldh4** [comment]
Fixed by #5590.
